<table align="center" width=100%>
    <tr>
        <td width="10%">
            <img src="in_class.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Inclass-Lab  <br>(Day 3)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

## Table of Content

1. **[Conditional Statements](#cond)**
2. **[Python Flow Control](#flow_control)**
3. **[List Comprehension](#lc)**



<a id="cond"> </a>
## 1. Conditional Statements

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Write a program to check whether a given number is multiple of 7 (use if statement)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [33]:
# type your code here

number = 7

if (number%7==0):
    print("Yes , it is multiple of 7 ")
else:
    print('this number is not a multiple of 7 ')

Yes , it is multiple of 7 


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. Write a code to print the maximum of two numbers (use if-else)</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [34]:
# type your code here

num1 = 20
num2 = 10
if num1>num2:
    print(f'{num1} is greater than other.')
else:
    print(f'{num2} is greater than other')

20 is greater than other.


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Write a program to check whether a number is positive, negative or zero (use if-elif) </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [35]:
# type your code here

num1 = int(input('Enter Any Number : '))

if (num1>0):
    print(f'This number is a Positive Number {num1}')
else:
    print(f'This number is a Negative Number {num1}')

Enter Any Number : 87
This number is a Positive Number 87


### Let's begin with some hands-on practice exercises

<a id= "flow_control"> </a>
## 2. Python Flow Control

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Write a code to print table of 5 using assignment operand +=.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [19]:
# type your code here

sum1 = 0 
for i in range(1,11):
    sum1+=5
    print(f'5 x {i} = {sum1}')

5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. Write a program for generating a fibonacci series starting with 0 and 1 for the next 10 values using a while loop. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [31]:
# type your code here  
count=0
n1=0
n2=1
print("Fibonacci sequence:")
while count < 10:
    print(n1 , end=" ")
    nth = n1 + n2
       # update values
    n1 = n2
    n2 = nth
    count += 1

Fibonacci sequence:
0 1 1 2 3 5 8 13 21 34 

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. Write a program to check whether a number is greater than the other using if statement (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [32]:
# type your code here   

num1 = int(input('Enter Any Number : '))
num2 = int(input('Enter Any Number : '))

if num1>num2:
    print(f'{num1} is greater than other.')
else:
    print(f'{num2} is greater than other')

Enter Any Number : 10
Enter Any Number : 20
20 is greater than other


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Write a code to check whether a number is divisible by 7 or not (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
# type your code here
number = int(input('Enter Any Number : '))

if (number%7==0):
    print("Yes , it is multiple of 7 ")
else:
    print('this number is not a multiple of 7 ')

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Write a code to find factorial of a number (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [43]:
# type your code here

num_fact = int(input("Enter any number : "))
factor = 1
for i in range(1 , num_fact+1):
    factor = factor * i
    

print(f'The factorial of of number {num_fact} : {factor}')

Enter any number : 6
The factorial of of number 6 : 720


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Write a program to check whether a number is prime or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [65]:
prime_num = int(input("enter any number : "))
if prime_num==1:
    print(f"this is prime number {prime_num}")
elif prime_num>1:
    for i in range(2 , prime_num):
        if prime_num%i==0:
            print(f'this is not a prime number {prime_num}')
            break
    else:
        print(f'this is prime number {i}')

enter any number : 5
this is prime number 4


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. Write a program to check whether two numbers are amicable or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [73]:
num1 = int(input("enter any number : "))
num2 = int(input("enter any number : "))

sum_num1 = 0
sum_num2 = 0

for i in range(1,num1):
    if num1%i==0:
        sum_num1+=i
for i in range(1,num2):
    if num2%i==0:
        sum_num2+=i
        
if(sum_num1==y and sum_num2==x):
    print("Amicable !")
else:
    print("Non-Amicable !")

enter any number : 112
enter any number : 112
Non-Amicable !


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. Reverse string using a for loop (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [79]:
strings = input("enter any string : ") 

new_string = " "
for i in strings:
    new_string = i+new_string
print(new_string)

enter any string : vikas chauhan
nahuahc sakiv 


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. Write a code to find the average of given numbers (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [89]:
# type your code here

iteration_of_number = int(input("Enter any number : "))
tt_list = []
for i in range(1, iteration_of_number+1):
    num_input = int(input(f"enter any number {i} : "))
    tt_list.append(num_input)
# print(tt_list)

total_tt_list = sum(tt_list)
# print(total_tt_list)

len_tt_list = len(tt_list)

avg_number = total_tt_list/len_tt_list
print(f'Average of Number is {avg_number}. And Sum , Count is {total_tt_list} , {len_tt_list}')

Enter any number : 10
enter any number 1 : 2
enter any number 2 : 2
enter any number 3 : 2
enter any number 4 : 2
enter any number 5 : 2
enter any number 6 : 2
enter any number 7 : 2
enter any number 8 : 2
enter any number 9 : 2
enter any number 10 : 200
Average of Number is 21.8. And Sum , Count is 218 , 10


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. Write a program to find the area of a circle for a given radius (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [92]:
# type your code here
pi = 3.14159265359

radius = int(input("Enter any number for radius : "))

area = pi * (radius*radius)
print(f'Area of circle with radius {radius} is {area}')

Enter any number for radius : 10
Area of circle with radius 10 is 314.159265359


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. Write a code to find the simple interest (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [95]:
# type your code here

principle = int(input("enter principle amount : "))
time_taken = int(input("enter time taken by person  in yrs : "))
rate_of_int = int(input("enter rate of interest in % : "))

si = (principle*time_taken*rate_of_int)/100

print(f'Simple interest of given detail is {si}')

enter principle amount : 1000
enter time taken by person  in yrs : 5
enter rate of interest in % : 1
Simple interest of given detail is 50.0


<a id= "lc"> </a>
## 3. List Comprehension

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Use list comprehension to obtain the squre root of first 10 natural numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [100]:
# type your code here

import math

sqrt_num = [round(x**0.5,2)for i in range(1,10)]
print(sqrt_num)

[1.0, 1.4142135623730951, 1.7320508075688772, 2.0, 2.23606797749979, 2.449489742783178, 2.6457513110645907, 2.8284271247461903, 3.0]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Use list comprehension to find the used vowels in a given sentence and/or word (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [101]:
# type your code here

sen=input("enter a sentence ")
[char for char in sen if char in ['a','e','i','o','u','A','E','I','O','U']]


enter a sentence vikas chauhan


['i', 'a', 'a', 'u', 'a']

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>17. Create a dictionary and access it values using a condition on its key. The data is given below. Let the condidtion on the key is that it should be a even number.</b>
                </font>
            </div>
       </td>
    </tr>
</table>


<table>
<tr>
    <th>Key</th>
    <td>1</td>
    <td>2</td>
    <td>3</td>
    <td>4</td> 
    <td>5</td> 
</tr>
<tr>
    <th>Name</th>
    <td>Aman</td>
    <td>Mohit</td>
    <td>Guari</td>
    <td>Imran</td>
    <td>Roma</td
     </tr>
<tr>
    <th>Marks</th>
    <td>24</td>
    <td>25</td>
    <td>26</td>
    <td>24</td>
    <td>27</td>
   </tr>
</table>

In [102]:
# type your code here
dict1={1:{'Aman':24},
       2:{'Mohit':25},
       3:{'Guari':26},
       4:{'Imran':24},
       5:{'Roma':27}}
print(dict1)

{1: {'Aman': 24}, 2: {'Mohit': 25}, 3: {'Guari': 26}, 4: {'Imran': 24}, 5: {'Roma': 27}}


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. Use list comprehension to find even and odd numbers from first 20 whole numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [108]:
# type your code here
even = [i for i in range(21) if i%2==0]
print(f'Even number list : {even}')

odd = [i for i in range(21) if i%2!=0]
print(f'Odd number list : {odd}')

Even number list : [0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20]
Odd number list : [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. Use list comprehension to print numbers divisible  by 2 and 3 in between 1 and 100. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [110]:
# type your code here

div_2_3 = [x for x in range(100) if x%2==0 and x%3==0]
print(f'Divisible by 2 and 3 : {div_2_3}')


Divisible by 2 and 3 : [0, 6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Use list comprehension to create a dictionary such that its keys are numbers from 1 to 10 and values are the corresponding to that key are its cube.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [111]:
# type your code here

dict_cube_create = {k:k**3 for k in range(11)}
print(dict_cube_create)

{0: 0, 1: 1, 2: 8, 3: 27, 4: 64, 5: 125, 6: 216, 7: 343, 8: 512, 9: 729, 10: 1000}


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>21. Use list comprehension to extract numbers from a string </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [112]:
# type your code here

xyz = input("Enter Any Number : ")

string  = [x for x in xyz if x.isdigit()]
print(string)

Enter Any Number : vikas chauhan19971214
['1', '9', '9', '7', '1', '2', '1', '4']


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>22. Use list comprehension to print table of 11 to 20 as shown below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
<tr>
    <td>11</td><td>22</td><td>33</td><td>44</td><td>55</td><td>66</td><td>77</td><td>88</td><td>99</td><td>110</td> 
</tr>
    
<tr>
    <td>12</td><td>24</td><td>36</td><td>48</td><td>60</td><td>72</td><td>84</td><td>96</td><td>108</td><td>120</td> 
</tr>

<tr>
    <td>13</td><td>36</td><td>39</td><td>52</td><td>65</td><td>78</td><td>91</td><td>104</td><td>117</td><td>130</td> 
</tr> 

<tr>
     <td>14</td><td>28</td><td>42</td><td>56</td><td>70</td> <td>84</td><td>98</td><td>112</td><td>126</td><td>140</td> 
</tr>

<tr>
     <td>15</td><td>30</td><td>45</td><td>60</td><td>75</td><td>90</td><td>105</td><td>120</td><td>135</td><td>150</td> 
</tr>

<tr>
     <td>16</td><td>32</td><td>48</td><td>64</td><td>80</td><td>96</td><td>112</td><td>128</td><td>144</td><td>160</td> 
</tr> 

<tr>
     <td>17</td><td>34</td><td>51</td><td>68</td><td>85</td><td>102</td><td>119</td><td>136</td><td>153</td><td>170</td> 
</tr> 

<tr>
     <td>18</td><td>36</td><td>54</td><td>72</td><td>90</td> <td>108</td><td>126</td><td>144</td><td>162</td><td>180</td> 
</tr>

<tr>
 <td>19</td><td>38</td><td>57</td><td>76</td><td>95</td><td>114</td><td>133</td><td>152</td><td>171</td><td>190</td> 
</tr>

<tr>
     <td>20</td><td>40</td><td>60</td><td>80</td><td>100</td><td>120</td><td>140</td><td>160</td><td>180</td><td>200</td> 
<tr>    
    
</table>

In [115]:
# type your code here

table_11_20 = [[k*n for k in range(1,11)] for n in range(11,21)]
table_11_20

[[11, 22, 33, 44, 55, 66, 77, 88, 99, 110],
 [12, 24, 36, 48, 60, 72, 84, 96, 108, 120],
 [13, 26, 39, 52, 65, 78, 91, 104, 117, 130],
 [14, 28, 42, 56, 70, 84, 98, 112, 126, 140],
 [15, 30, 45, 60, 75, 90, 105, 120, 135, 150],
 [16, 32, 48, 64, 80, 96, 112, 128, 144, 160],
 [17, 34, 51, 68, 85, 102, 119, 136, 153, 170],
 [18, 36, 54, 72, 90, 108, 126, 144, 162, 180],
 [19, 38, 57, 76, 95, 114, 133, 152, 171, 190],
 [20, 40, 60, 80, 100, 120, 140, 160, 180, 200]]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>23. Use list comprehension to find transpose of the matrix given below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
    <tr>
        <td>1</td>
        <td>2</td>
    </tr>
    <tr>
        <td>3</td>
        <td>4</td>
    </tr>
    <tr>
        <td>5</td>
        <td>6</td>
    </tr>
    <tr>
        <td>7</td>
        <td>8</td>
    </tr>
    </table>

In [116]:
# type your code here

matrix=[[1,2],[3,4],[5,6],[7,8]]
list1=[[row[i] for row in matrix] for i in range(2)]
list1

[[1, 3, 5, 7], [2, 4, 6, 8]]